### PR TITLE
URL encoding of query strings

### LINKED ISSUE #3: spaces in url query strings
In your GET example, the space in ($id: ID!) invalidates the request.
```
http://example.com/graphql?query=query($id: ID!){user(id:$id){name}}&variables={"id":"QVBJcy5ndXJ1"}
```
Do you recommend clients implementing a graphql api remove all spaces in queries or url encode the query string. For example:
```
url = 'http://example.com/graphql?query=' + quote('query($id: ID!){user(id:$id){name}}&variables={"id":"QVBJcy5ndXJ1"}')
```

### PR DESCRIPTION
**Pull request by benjie**
Fixes #3. Sorry about the delay.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,8 +107,27 @@ Note: specifying `null` in JSON (or equivalent values in other formats) as value
 ## GET
 
 For HTTP GET requests, the query parameters should be provided in the query component of the request URL in the form of
-`key=value` pairs with `&` symbol as a separator.
-The value of the `variables` parameter should be represented as a JSON-encoded string.
+`key=value` pairs with `&` symbol as a separator and both the key and value having their "reserved" characters percent-encoded as specified in [section 2 of RFC3986](https://tools.ietf.org/html/rfc3986#section-2).
+The unencoded value of the `variables` parameter should be represented as a JSON-encoded string.
+
+For example, a function in JavaScript to build the URL for an operation might look like:
```
Comment (IvanGoncharov) [inline README.md]: Not sure we should add JS code into spec.
I think the bellow example should be enough.

Comment (benjie) [reply README.md]: Is not an example implementation useful / in-scope?

Comment (benjie) [reply README.md]: (Happy to remove.)

Comment (spawnia) [reply README.md]: I see value in having code examples like this, although we might have them as part of a conformance test suite/reference implementation.

**Comment by benjie** [comment]
Maybe we should discuss the plan for code examples at the WG?

File: README.md [hunk 2]
```diff
@@ -107,8 +107,27 @@ Note: specifying `null` in JSON (or equivalent values in other formats) as value
 ## GET
 
 For HTTP GET requests, the query parameters should be provided in the query component of the request URL in the form of
-`key=value` pairs with `&` symbol as a separator.
-The value of the `variables` parameter should be represented as a JSON-encoded string.
+`key=value` pairs with `&` symbol as a separator and both the key and value having their "reserved" characters percent-encoded as specified in [section 2 of RFC3986](https://tools.ietf.org/html/rfc3986#section-2).
```
Comment (benjie) [inline README.md]:
```suggestion
`key=value` pairs with `&` symbol as a separator and both the key and value should have their "reserved" characters percent-encoded as specified in [section 2 of RFC3986](https://tools.ietf.org/html/rfc3986#section-2).
```

**Comment by benjie** [comment]
@IvanGoncharov Ready for re-review

**Comment by benjie** [comment]
<details>
<summary>Removed code sample</summary>

```js
function getURLForOperation(query, variables, operationName) {
  let queryString = `query=${encodeURIComponent(query)}`;
  if (variables) {
    queryString += `&variables=${encodeURIComponent(
      JSON.stringify(variables)
    )}`;
  }
  if (operationName) {
    queryString += `&operationName=${encodeURIComponent(
      operationName
    )}`;
  }
  return `/graphql?${queryString}`;
}
```

</details>

**Pull request merged**